### PR TITLE
feat(frontend): BTC transaction stores

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

Add BTC transaction stores. It will be later extended with additional methods, similar to other related stores (e.g. `cleanUp`).

### CODE REVIEW

File: src/frontend/src/btc/stores/btc-transactions.store.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import type { CertifiedData } from '$lib/types/store';
+import type { TokenId } from '$lib/types/token';
+import { nonNullish } from '@dfinity/utils';
+
+export type BtcCertifiedTransaction = CertifiedData<BtcTransactionUi>;
+
+export type BtcTransactionsData = BtcCertifiedTransaction[];
+
+export interface BtcTransactionsStore extends CertifiedStore<BtcTransactionsData> {
+	prepend: (params: { tokenId: TokenId; transactions: BtcCertifiedTransaction[] }) => void;
```
Comment (peterpeterparker) [inline src/frontend/src/btc/stores/btc-transactions.store.ts:12]: No append and no cleanUp?

File: src/frontend/src/btc/stores/btc-transactions.store.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+const initBtcTransactionsStore = (): BtcTransactionsStore => {
+	const { subscribe, update, reset } = initCertifiedStore<BtcTransactionsData>();
+
+	return {
+		prepend: ({
```
Comment (peterpeterparker) [inline src/frontend/src/btc/stores/btc-transactions.store.ts:19]: Is it the exact same code as IC transactions stores?

File: src/frontend/src/btc/stores/btc-transactions.store.ts (hunk 1 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/btc/stores/btc-transactions.store.ts]: It now uses a comment store template, so both append and cleanUp are a part of btc transactions store.

File: src/frontend/src/btc/stores/btc-transactions.store.ts (hunk 2 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/btc/stores/btc-transactions.store.ts]: With a small adjustment to BtcTransactionUi type (in another PR), the stores are now 100% identical. I extracted it in a separate store in `lib`. BTC already uses it in this PR. And for IC, I'll refactor the code separately after we merge this feature branch to main.

File: src/frontend/src/lib/stores/transactions.store.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+export type TransactionsData<T> = CertifiedTransaction<T>[];
+
+export interface TransactionsStore<T> extends CertifiedStore<TransactionsData<T>> {
+	prepend: (params: { tokenId: TokenId; transactions: CertifiedTransaction<T>[] }) => void;
+	append: (params: { tokenId: TokenId; transactions: CertifiedTransaction<T>[] }) => void;
+	cleanUp: (params: { tokenId: TokenId; transactionIds: string[] }) => void;
+}
+
+export const initTransactionsStore = <
```
Comment (peterpeterparker) [inline src/frontend/src/lib/stores/transactions.store.ts:18]: I assume this is "basically" a refactoring of existing code and we will use it for IC in a separate PR right?

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

File: src/frontend/src/lib/stores/transactions.store.ts (hunk 3 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/stores/transactions.store.ts:18]: Exactly.

**Pull request merged**